WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionFilters.js

```diff
@@ -0,0 +1,591 @@
+import React, { useState, useEffect, useReducer } from 'react';
+import { useInView } from 'react-intersection-observer';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import {
+  Button, Icon, Form, Container, Dropdown, Popup, List, Input, Header,
+} from 'semantic-ui-react';
+import DatePicker from 'react-datepicker';
+
+import I18n from '../../lib/i18n';
+import { competitionsApiUrl, delegatesApiUrl } from '../../lib/requests/routes.js.erb';
+import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken';
+import {
+  events, continents, countries, competitionConstants,
+} from '../../lib/wca-data.js.erb';
+
+import CompetitionTable from './CompetitionTable';
+import CompetitionMap from './CompetitionMap';
+
+import 'react-datepicker/dist/react-datepicker.css';
+
+// Max number of competitions fetched per query
+const COMPETITIONS_API_PAGINATION = 25;
+
+// Limit number of markers on map, especially for "All Past Competitions"
+const MAP_DISPLAY_LIMIT = 500;
+
+const WCA_EVENT_IDS = Object.values(events.official).map((e) => e.id);
+
+const PAST_YEARS_WITH_COMPETITIONS = [];
+for (let year = new Date().getFullYear(); year >= 2003; year -= 1) {
+  PAST_YEARS_WITH_COMPETITIONS.push(year);
+}
+PAST_YEARS_WITH_COMPETITIONS.push(1982);
+
+const regionsOptions = [
+  { key: 'all', text: I18n.t('common.all_regions'), value: 'all_regions' },
+  {
+    key: 'continents_header', value: '', disabled: true, content: <Header content={I18n.t('common.continent')} size="small" style={{ textAlign: 'center' }} />,
+  },
+  ...(Object.values(continents.real).map((continent) => (
+    { key: continent.id, text: continent.name, value: continent.id }
+  ))),
+  {
+    key: 'countries_header', value: '', disabled: true, content: <Header content={I18n.t('common.country')} size="small" style={{ textAlign: 'center' }} />,
+  },
+  ...(Object.values(countries.real).map((country) => (
+    { key: country.id, text: country.name, value: country.iso2 }
+  ))),
+];
+
+const filterInitialState = {
+  timeOrder: 'present',
+  selectedYear: 'all_years',
+  customStartDate: null,
+  customEndDate: null,
+  region: 'all_regions',
+  delegate: '',
+  search: '',
+};
+
+const calculateQueryKey = (filterState) => {
+  let timeKey = '';
+  if (filterState?.timeOrder === 'past') {
+    timeKey = `${filterState.selectedYear}`;
+  } else if (filterState?.timeOrder === 'custom') {
+    timeKey = `start${filterState.customStartDate}-end${filterState.customEndDate}`;
+  }
+
+  return {
+    timeOrder: filterState?.timeOrder,
+    region: filterState?.region,
+    delegate: filterState?.delegate,
+    search: filterState?.search,
+    time: timeKey,
+  };
+};
+
+function CompetitionFilters() {
+  const [selectedEvents, setSelectedEvents] = useState([]);
+  const [shouldShowRegStatus, setShouldShowRegStatus] = useState(false);
+  const [shouldShowCancelled, setShouldShowCancelled] = useState(false);
```

Comment:
To differentiate the fact that should _show_ registration status is about showing additional information vs should _include_ cancelled is about the filter logic.
```suggestion
  const [shouldIncludeCancelled, setShouldIncludeCancelled] = useState(false);
```